diff --git a/node_modules/lottie-react-native/src/android/build.gradle b/node_modules/lottie-react-native/src/android/build.gradle
index 1234567..abcdefg 100644
--- a/node_modules/lottie-react-native/src/android/build.gradle
+++ b/node_modules/lottie-react-native/src/android/build.gradle
@@ -7,6 +7,7 @@ apply plugin: 'com.android.library'
 //def DEFAULT_ANDROID_SUPPORT_LIB_VERSION = "1.0.0"
 
 android {
+  namespace "com.airbnb.android.react.lottie"
   compileSdkVersion safeExtGet('compileSdkVersion', 30)
 
   defaultConfig {
